### PR TITLE
Turns on dns horizontal scaling tests for GKE

### PR DESCRIPTION
**Pull request by MrHohn**
Seems like the dns-autoscaler is already enabled in [this recent gke build](https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/ci-kubernetes-e2e-gci-gke/769/).
Turning on the corresponding e2e tests to increase test coverage.

Probably better to wait for this fix #37261 to go in first.

@bowei @bprashanth 
cc @maisem @roberthbailey

### CODE REVIEW

**Comment by k8s-oncall** [comment]
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/37850)

**Comment by MrHohn** [comment]
@k8s-bot gce etcd3 e2e test this, issue: #37213

**Comment by MrHohn** [comment]
cc @thockin

File: cluster/gke/config-default.sh [hunk 1]
```diff
@@ -42,6 +42,6 @@ ENABLE_L7_LOADBALANCING="${KUBE_ENABLE_L7_LOADBALANCING:-glbc}"
 ENABLE_CLUSTER_MONITORING="${KUBE_ENABLE_CLUSTER_MONITORING:-standalone}"
 
 # Optional: Enable DNS horizontal autoscaler
-ENABLE_DNS_HORIZONTAL_AUTOSCALER="${KUBE_ENABLE_DNS_HORIZONTAL_AUTOSCALER:-false}"
+ENABLE_DNS_HORIZONTAL_AUTOSCALER="${KUBE_ENABLE_DNS_HORIZONTAL_AUTOSCALER:-true}"
```
Comment (roberthbailey) [inline cluster/gke/config-default.sh]: Setting this doesn't actually have an effect (regardless of the value).

**Comment by roberthbailey** [comment]
the e2e change lgtm.

File: cluster/gke/config-default.sh (hunk 1 above)
Comment (bprashanth) [inline cluster/gke/config-default.sh]: does this need to be piped through elsewhere internally, or is this one env var enough?

**Comment by bprashanth** [comment]
LGTM just the one question

File: cluster/gke/config-default.sh (hunk 1 above)
Comment (MrHohn) [inline cluster/gke/config-default.sh]: This env is not enough, but the real env was already piped into GKE internally.

Comment (bprashanth) [inline cluster/gke/config-default.sh]: yeah someone might be fooled into thinking they can turn it off by flipping it

Comment (MrHohn) [inline cluster/gke/config-default.sh]: Sure, I'm removing it.

**Comment by MrHohn** [comment]
The dummy env for GKE is removed.

**Pull request merged**